### ISSUE TITLE
xLights/outputs/serial_posix.cpp  - typo?

### ISSUE THREAD

**Issue by cybie257** [smeighan/xLights#1280]
Compiled latest on Linux and came across a failed build on errors in the xLights/outputs/serial_posix.cpp file. 

void SerialPort::SetRTS(bool state)
{
    int RTS_flag = TIOCM_RTS;
    if (state)
    {
        ioctl(fd, TIOCMBIS, &RTS_flag);
    }
    else
    {
        ioctl(fd, TIOCMBIC, &RTS_flag);
    }
};

Seems the correction is as simple as adding Underscores before the "fd" (IE : _fd) as the compiler was complaining about fd not being declared with suggested _fd. 

After adding the underscores, the errors were no longer present and build successful. 

-Tim (aka Cybie257)

**Comment by AzGilrock** [comment]
Well if you pull code that is in progress since the last release you're going to run into this.  There is a Travis build for Linux that flags these for us so they get fixed before the official releases are done.  I would only pull up to the latest release checkin if you are doing your own builds.

**Comment by cjd** [comment]
If you re-pull now you will get a fixed version.
There is the ability for git to pull at a specific tag - but that won't always help as sometimes the release-tagged version needs a quick fix to work properly on linux (this happens less often now with Travis doing auto builds)
Is there a reason you are manually pulling/building?
If you are not doing development then using the pre-built appimage is preferred

**Comment by cybie257** [comment]
Hi Chris,

Thank you. The reason I have been building from source is due to t he previous issues with text and/or Gif crashed I experienced along with another user. No matter the Linux Distro or version I would use, the Repo or AppImage would seem to crash when selecting OS Fonts and animated Gifs. When I compiled from source, those issues went away and I had a much more stable experience. I may try the AppImage again and see how it goes. Thanks again!
